lib/galaxy/tools/imp_exp/__init__.py

```diff
@@ -150,26 +140,22 @@ def get_tag_str(tag, value):
                             os.path.realpath(os.path.abspath(os.path.join(archive_dir, dataset_attrs['file_name'])))
                         if not file_in_dir(temp_dataset_file_name, os.path.join(archive_dir, "datasets")):
                             raise MalformedContents("Invalid dataset path: %s" % temp_dataset_file_name)
-                        if datasets_usage_counts[temp_dataset_file_name] == 1:
-                            self.app.object_store.update_from_file(hda.dataset, file_name=temp_dataset_file_name, create=True)
-
-                            # Import additional files if present. Histories exported previously might not have this attribute set.
-                            dataset_extra_files_path = dataset_attrs.get('extra_files_path', None)
-                            if dataset_extra_files_path:
-                                try:
-                                    file_list = os.listdir(os.path.join(archive_dir, dataset_extra_files_path))
-                                except OSError:
-                                    file_list = []
-
-                                if file_list:
-                                    for extra_file in file_list:
-                                        self.app.object_store.update_from_file(
-                                            hda.dataset, extra_dir='dataset_%s_files' % hda.dataset.id,
-                                            alt_name=extra_file, file_name=os.path.join(archive_dir, dataset_extra_files_path, extra_file),
-                                            create=True)
-                        else:
-                            datasets_usage_counts[temp_dataset_file_name] -= 1
-                            shutil.copyfile(temp_dataset_file_name, hda.file_name)
+                        self.app.object_store.update_from_file(hda.dataset, file_name=temp_dataset_file_name, create=True)
```

Comment:
This was broken because it creates HDAs with associated `Dataset` instances without actually creating the file for the `Dataset` instance in the object store, right ? So would it make sense to do something like:

```
imported_datasets = {}
...
if not temp_dataset_file_name in imported_datasets:
   <all the necessary stuff>
    imported_datasets[temp_dataset_file_name] = hda.dataset
else:
    hda.dataset = imported_datasets[temp_dataset_file_name]
```
?

That would avoid importing HDA copies multiple times, I think.

Reply:
But with if ``extra_files_path`` doesn't match - I feel like this never happens and we should just simplify the code for now with this fix and make this optimization down the road with tests to verify everything works.

Reply:
Ah, right. Fair enough!